zips/zip-0228.rst

```diff
@@ -2,12 +2,539 @@
+Selecting an appropriate matching policy is key, as it heavily influences the trading experience on the system. 
+Opting for a rigid policy may either lead to a poorly used protocol (users staying on existing trading venues) or may force users to interact with the protocol in a way that harms and compromises their on-chain privacy (e.g., carrying out funds consolidation on chain to trade via the Matcher, leaking lots of information in the process). 
+As such, designing a protocol with flexibility in mind is key to minimize information leakage, but it is also fundamental for liquidity consolidation on the system. 
+
+
+Limitation: Partial Fills
+-------------------------
+
+In this ZIP, **we assume that orders are matched perfectly**, and we do not support partial fills.
+In practice though, modern CLOBs offer partial fills to their users, allowing trades with matching prices to be matched even if quantities don't align perfectly.
+Offering partial fills without making strong assumptions on the Matcher (e.g. “The Matcher is a market maker that takes the other side of each trade”) or on the availability of traders (e.g. “Traders stay online and stay connected to the Matcher to split their orders on-demand”) is hard.
+This is particularly at odds with the protocol in this ZIP, since the Matcher is non-custodian and thus only manipulates commitments and notes, which are generated before the matching occurs: traders commit to values, which may need to be changed - in the future - to align with the matching operation of the Matcher (done at a subsequent time).
+
+Some of the proposals to allow for this are:
+
+- The use of standard denomination for orders.
+- The inclusion of a special flag “partialFill = true” to flag to the Matcher that the trader is willing to stay online to split orders on demand.
+- Existence of a third party Market Maker providing liquidity to the many ZSA pairs traded on the system.
+
+All of these proposals have their pros and cons, and warrant further study and discussion.
+
+Fairness
+--------
+
+It is important to provide a good execution for both sides of a Swap. 
+As such, one side should not have an advantage against the other. 
+Every trading party must be able to withdraw their trading intent (i.e., cancel their orders).
+The main considerations here are:
+
+- Provide a mechanism for time-limiting Swaps to reduce freezing up of Assets due to their being locked up in Swaps that have not been completed, and to align with dynamically changing “exchange rates” between different Assets.
+- Consider repercussions of, and possible mitigations for, the “free option” that becomes available to the party that performs the last step of the protocol. That is, it is possible for that party to abort the protocol at this last step if the Swap is no longer beneficial for it.
```

Comment:
This section is very vague and repeats previously mentioned ideas. Rewrite into something more useful or remove completely.

Reply:
I will remove it, we have covered the material elsewhere.